composer/trainer/trainer.py

```diff
@@ -1515,21 +1507,25 @@ def _train_loop(self) -> None:
                         scheduler.step()
 
                 self.engine.run_event(Event.EPOCH_END)
-
-                for evaluator in self.state.evaluators:
-                    assert evaluator.eval_interval is not None, "eval_interval should have been set on __init__() or fit()"
-                    assert evaluator.subset_num_batches is not None, "subset_num_batches should have been set on __init__() or fit()"
-                    if evaluator.eval_interval(self.state, Event.EPOCH_END):
-                        self.eval(
-                            dataloader=evaluator.dataloader,
-                            dataloader_label=evaluator.label,
-                            subset_num_batches=evaluator.subset_num_batches,
-                            metrics=evaluator.metrics,
-                            log_level=LogLevel.EPOCH,
-                        )
-
+                self._run_evaluators(Event.EPOCH_END)
                 self.engine.run_event(Event.EPOCH_CHECKPOINT)
         self.engine.run_event(Event.FIT_END)
+        self._run_evaluators(Event.FIT_END)
+
+    def _run_evaluators(self, event: Event):
+        """Runs evaluators periodically during training."""
+
+        for evaluator in self.state.evaluators:
+            assert evaluator.eval_interval is not None, "eval_interval should have been set on __init__() or fit()"
+            assert evaluator.subset_num_batches is not None, "subset_num_batches should have been set on __init__() or fit()"
+            if evaluator.eval_interval(self.state, event):
+                self.eval(
+                    dataloader=evaluator.dataloader,
+                    dataloader_label=evaluator.label,
+                    subset_num_batches=evaluator.subset_num_batches,
+                    metrics=evaluator.metrics,
+                    log_level=LogLevel.EPOCH,
```

Comment:
Can you update the `log_level` here to match the event? E.g. epoch_end -> LogLevel.EPOCH; batch_end -> LogLevel.BATCH; fit_end -> LogLevel.FIT

Reply:
Done!